.github/scripts/post_to_bluesky.py

```diff
@@ -0,0 +1,228 @@
+    
+    def post_message(self, message):
+        """Post a message to Bluesky"""
+        if not self.session:
+            logger.error("Not authenticated. Call authenticate() first.")
+            return False
+            
+        try:
+            post_url = f"{self.base_url}/xrpc/com.atproto.repo.createRecord"
+            headers = {
+                "Authorization": f"Bearer {self.session['accessJwt']}",
+                "Content-Type": "application/json"
+            }
+            
+            # Create the post record
+            now = datetime.now(timezone.utc).isoformat()
+            post_data = {
+                "repo": self.session["did"],
+                "collection": "app.bsky.feed.post",
+                "record": {
+                    "$type": "app.bsky.feed.post",
+                    "text": message,
+                    "createdAt": now
+                }
+            }
+            
+            response = requests.post(post_url, headers=headers, json=post_data)
+            response.raise_for_status()
+            
+            logger.info("Successfully posted to Bluesky")
+            return True
+            
+        except requests.exceptions.RequestException as e:
+            logger.error(f"Failed to post to Bluesky: {e}")
+            return False
+
+class GoogleCalendarFetcher:
+    def __init__(self, api_key):
+        self.api_key = api_key
+        self.calendar_id = "ekol7ulqm14nv155angut2rlfo@group.calendar.google.com"
+        
+    def get_next_event(self):
+        """Fetch the next upcoming event from the Google Calendar"""
+        try:
+            # Current time in ISO format
+            time_min = datetime.now(timezone.utc).isoformat()
+            
+            url = "https://www.googleapis.com/calendar/v3/calendars/{}/events".format(self.calendar_id)
+            params = {
+                "key": self.api_key,
+                "timeMin": time_min,
+                "maxResults": 1,
+                "singleEvents": True,
+                "orderBy": "startTime"
+            }
+            
+            response = requests.get(url, params=params)
+            response.raise_for_status()
+            
+            data = response.json()
+            events = data.get("items", [])
+            
+            if not events:
+                logger.warning("No upcoming events found")
+                return None
+                
+            event = events[0]
+            logger.info(f"Found next event: {event.get('summary', 'Unknown')}")
+            return event
+            
+        except requests.exceptions.RequestException as e:
+            logger.error(f"Failed to fetch calendar events: {e}")
+            return None
+
+def format_event_message(event):
+    """Format the event data into a Bluesky post message"""
+    if not event:
+        return "Join us for our next Utah Data Science Center event! Check our website for details: https://datascience.utah.edu/seminar"
```

Comment:
If there's no event, then return an empty string. Then, no posting should be done.